### PR TITLE
api: unexport apitypes.Status fields

### PR DESCRIPTION
**Pull request by buck54321**
Getters and setters implemented. New apitypes.APIStatus type for /status response.

For #1094, #1150.

### CODE REVIEW

File: api/insight/apiroutes.go [hunk 1]
```diff
@@ -41,31 +40,22 @@ type insightApiContext struct {
...
 	ReqPerSecLimit float64
 	maxCSVAddrs    int
 }
 
 // NewInsightContext Constructor for insightApiContext
 func NewInsightContext(client *rpcclient.Client, blockData *dcrpg.ChainDBRPC, params *chaincfg.Params,
-	memPoolData DataSourceLite, JSONIndent string, maxAddrs int) *insightApiContext {
-	conns, _ := client.GetConnectionCount()
-	nodeHeight, _ := client.GetBlockCount()
-	version := semver.NewSemver(1, 0, 0)
```
Comment (buck54321) [inline api/insight/apiroutes.go]: This version appeared to be unused.

Comment (chappjc) [reply api/insight/apiroutes.go]: I don't think it's useful here.  Good to remove.

File: api/types/apitypes.go [hunk 2]
```diff
@@ -421,11 +431,78 @@ type Status struct {
 	NetworkName     string `json:"network_name"`
 }
 
-// GetHeight returns the last stored block height.
-func (s *Status) GetHeight() uint32 {
+// NewStatus is the constructor for a new Status.
+func NewStatus(nodeHeight uint32, conns int64, apiVersion int, dcrdataVersion, netName string) *Status {
+	return &Status{
+		RWMutex:         sync.RWMutex{},
```
Comment (chappjc) [inline api/types/apitypes.go]: I'd prefer to omit this line since the zero value is assigned automatically.

File: api/types/apitypes.go [hunk 3]
```diff
@@ -421,11 +431,78 @@ type Status struct {
...
+// if Status.height is the same as Status.dbHeight.
+func (s *Status) SetHeight(height uint32) {
+	s.Lock()
+	defer s.Unlock()
+	s.ready = height == s.dbHeight
+	s.height = height
+}
+
+// SetHeightAndConnections simultaneously sets the node height and node
+// and connection count. Status.ready is set to true if the height and dbHeight
```
Comment (chappjc) [inline api/types/apitypes.go]: extra "and"

File: middleware/apimiddleware.go [hunk 4]
```diff
@@ -745,11 +745,6 @@ func BlockIndexLatestCtx(r *http.Request, source DataSource) context.Context {
 	return context.WithValue(r.Context(), ctxBlockIndex, int(idx)) // Must be int!
 }
 
-// StatusCtx embeds the specified apitypes.Status into a request context.
-func StatusCtx(r *http.Request, status *apitypes.Status) context.Context {
-	return context.WithValue(r.Context(), ctxAPIStatus, status)
-}
```
Comment (chappjc) [inline middleware/apimiddleware.go]: Let's nuke the `middleware.ctxAPIStatus` `contextKey` too.

**Pull request merged**